{
  if (err.get() != null)   fail("Exception has been caught by listener: " + err.get().getMessage());
}
